lib/routes/auth.js

```diff
@@ -44,4 +48,43 @@ router.post(
   }
 );
 
+// /api/auth/login
+router.post(
+  './login',
+  [
+    check('email', 'This email address is not registered')
+      .normalizeEmail()
+      .isEmail(),
+    check('password', 'Invalid password').exists(),
+  ],
+  async (request, response) => {
+    try {
+      const errors = validationResult(request);
+      if (!errors.isEmpty) {
+        return response.status(400).json({
+          errors: errors.array(),
+          message: 'Invalid log in data',
+        });
+      }
+
+      const { email, password } = request.body;
+      const user = await User.findOne({ email: email });
+      if (!user) {
+        return response.status(400).json({ message: 'User not found' });
+      }
+
+      const passwordCheck = await bcrypt.compare(password, user.password);
```

Comment:
`passwordCheck` sounds like a function.
`!passwordCheck` sounds like you didn't check for a password.

Maybe call it `passwordIsEqual` or `isCorrectPassword`

Reply:
Change it to `matchPassword`